Source/WebKit/WebProcess/WebPage/WebPage.cpp

```diff
@@ -8876,20 +8876,26 @@ bool WebPage::isUsingUISideCompositing() const
 void WebPage::setLinkDecorationFilteringData(Vector<WebCore::LinkDecorationFilteringData>&& strings)
 {
     m_linkDecorationFilteringData.clear();
-    m_domainScopedLinkDecorationFilteringData.clear();
 
     for (auto& data : strings) {
-        if (data.domain.isEmpty()) {
-            m_linkDecorationFilteringData.add(data.linkDecoration);
+        if (!m_linkDecorationFilteringData.isValidKey(data.linkDecoration)) {
+            ASSERT_NOT_REACHED();
             continue;
         }
 
-        if (!m_domainScopedLinkDecorationFilteringData.isValidKey(data.domain))
-            continue;
+        auto it = m_linkDecorationFilteringData.ensure(data.linkDecoration, [] {
+            return LinkDecorationFilteringConditionals { };
+        }).iterator;
 
-        m_domainScopedLinkDecorationFilteringData.ensure(data.domain, [&] {
-            return HashSet<String> { };
-        }).iterator->value.add(data.linkDecoration);
+        if (!data.domain.isEmpty()) {
+            if (auto& domains = it->value.domains; domains.isValidValue(data.domain))
+                domains.add(data.domain);
+            else
+                ASSERT_NOT_REACHED();
```

Comment:
And what's the correct behavior here? Should we just skip the domain and continue on to checking the path, or should we consider the data corrupt and do something else?

Reply:
I guess we could early return here, I don't have a preference. I just wanted these assertions to be hit if something was misconfigured when running the API tests.